staging/src/k8s.io/apiserver/pkg/storage/etcd3/feature/feature_support_checker_test.go

```diff
@@ -0,0 +1,283 @@
+					{Version: "3.4.30", Endpoint: "localhost:2390"}},
+					expectedResult: false}},
+		},
+		{
+			testName: "Disabled supported versions bound, 3.5.0",
+			rounds: []testCase{
+				{endpointsVersion: []mockEndpointVersion{
+					{Version: "3.5.0", Endpoint: "localhost:2390"}},
+					expectedResult: false}},
+		},
+		{
+			testName: "Disabled supported versions bound, 3.5.12",
+			rounds: []testCase{
+				{endpointsVersion: []mockEndpointVersion{
+					{Version: "3.5.12", Endpoint: "localhost:2390"}},
+					expectedResult: false}},
+		},
+		{
+			testName: "Disables if called with one client doesn't support it",
+			rounds: []testCase{
+				{endpointsVersion: []mockEndpointVersion{
+					{Version: "3.5.13", Endpoint: "localhost:2390"},
+					{Version: "3.5.10", Endpoint: "localhost:2391"}},
+					expectedResult: false}},
+		},
+		{
+			testName: "Disables if called with all client doesn't support it",
+			rounds: []testCase{
+				{endpointsVersion: []mockEndpointVersion{
+					{Version: "3.5.9", Endpoint: "localhost:2390"},
+					{Version: "3.5.10", Endpoint: "localhost:2391"}},
+					expectedResult: false}},
+		},
+		{
+			testName: "Enabled if provided client has at least one endpoint that supports it and no client that doesn't",
+			rounds: []testCase{
+				{endpointsVersion: []mockEndpointVersion{
+					{Version: "3.4.31", Endpoint: "localhost:2390"},
+					{Version: "3.5.13", Endpoint: "localhost:2391"},
+					{Version: "3.5.14", Endpoint: "localhost:2392"},
+					{Version: "3.6.0", Endpoint: "localhost:2393"}},
+					expectedResult: true}},
+		},
+		{
+			testName: "Disabled cannot be re-enabled",
+			rounds: []testCase{
+				{endpointsVersion: []mockEndpointVersion{
+					{Version: "3.4.0", Endpoint: "localhost:2390"},
+					{Version: "3.4.1", Endpoint: "localhost:2391"}},
+					expectedResult: false},
+				{endpointsVersion: []mockEndpointVersion{
+					{Version: "3.6.0", Endpoint: "localhost:2392"}},
+					expectedResult: false}},
+		},
+		{
+			testName: "Enabled after with one client that supports it, and later disabled it with second client that doesn't",
+			rounds: []testCase{
+				{endpointsVersion: []mockEndpointVersion{
+					{Version: "3.6.0", Endpoint: "localhost:2390"},
+					{Version: "3.5.14", Endpoint: "localhost:2391"}},
+					expectedResult: true},
+				{endpointsVersion: []mockEndpointVersion{
+					{Version: "3.4.0", Endpoint: "localhost:2392"}},
+					expectedResult: false}},
+		},
+		{
+			testName: "Disabled malformed version first client, success on second client",
+			rounds: []testCase{
+				{endpointsVersion: []mockEndpointVersion{
+					{Version: "3.5.--a", Endpoint: "localhost:2390"}},
+					expectedResult: false,
+				},
+				{endpointsVersion: []mockEndpointVersion{
+					{Version: "3.6.0", Endpoint: "localhost:2392"}},
+					expectedResult: false}},
+		},
+		{
+			testName: "Disabled success on first client, malformed version in one of second client's endpoints",
```

Comment:
```suggestion
			testName: "Disabled - success on first client, malformed version in one of second client's endpoints",
```
I see have a convention in naming, first what is the result of the test, and then description. That's good, however test name should be treated as sentence. Having a separator should make it clearer.

Reply:
Thanks, Done.